src/mame/roland/roland_d70.cpp

```diff
@@ -0,0 +1,562 @@
+  //  the PCM sample tables.
+  // All this is just for making debugging easier, as it allows one to check the
+  // register state using the Memory Viewer.
+  if (offset == 0x01 || offset == 0x21)
+    offset ^= 0x20; // remove when PCM data readback via sound chip is confirmed
+                    // to work
+  if (offset < 0x20)
+    return pcm->read(offset);
+  if (offset < 0x40)
+    offset -= 0x20;
+
+  if (offset == 0x01) {
+    // code for reading from the PCM sample table is at 0xB027
+    // The code at 0xB0AC writes to 1411/1F (??), then 1403/02 (bank), then
+    // 1409/08/0B/0A (address). It waits a few cycles and at 0xB0F7 it reads the
+    // resulting data from 1401.
+    offs_t bank = sound_io_buffer[0x03];
+    offs_t addr = (sound_io_buffer[0x09] << 0) | (sound_io_buffer[0x0A] << 8) |
+                  (sound_io_buffer[0x0B] << 16);
+    addr = ((addr >> 6) + 2) & 0x3FFFF;
+    addr |= (bank << 16);
+    // write actual ROM address to 1440..1443 for debugging
+    sound_io_buffer[0x43] = (addr >> 0) & 0xFF;
+    sound_io_buffer[0x42] = (addr >> 8) & 0xFF;
+    sound_io_buffer[0x41] = (addr >> 16) & 0xFF;
+    sound_io_buffer[0x40] = (addr >> 24) & 0xFF;
+    return pcmrom_r(addr);
+  }
+  return sound_io_buffer[offset];
+}
+
+void roland_d70_state::snd_io_w(offs_t offset, u8 data) {
+  // register map
+  // ------------
+  // Note: 16-bit words are Little Endian, the firmware writes the odd byte is
+  // first
+  //  00/01 - ??
+  //  02/03 - ROM bank (only bits 11-13 are used, bit 11 = PCM card, bits 12-13
+  //  select between IC18/19/20) 04/05 - frequency (2.14 fixed point, 0x4000 =
+  //  32000 Hz) 06/07 - volume 08/09 - sample start address, fraction (2.14
+  //  fixed point, i.e. 1 byte = 0x4000) 0A/0B - sample start address (high
+  //  word, i.e. address bits 2..17) 0C/0D - sample end address (high word)
+  //  0E/0F - sample loop address (high word)
+  //  11/13/15/17 - voice enable mask (11 = least significant 8 bits, 17 = most
+  //  significant 8 bits) 1A - ?? 1F - voice select
+  if (offset < 0x20)
+    pcm->write(offset, data);
+  sound_io_buffer[offset] = data;
+}
+
+u8 roland_d70_state::ach0_r() { return 128; }
+u8 roland_d70_state::ach1_r() { return 128; }
+u8 roland_d70_state::ach2_r() { return 128; }
+u8 roland_d70_state::ach3_r() { return 128; }
+u8 roland_d70_state::ach4_r() { return 128; }
+
+TIMER_DEVICE_CALLBACK_MEMBER(roland_d70_state::samples_timer_cb) {
+  // TODO: does this trigger something?
+}
+
+void roland_d70_state::lcd_palette(palette_device &palette) const {
+  palette.set_pen_color(0, rgb_t(138, 146, 148));
+  palette.set_pen_color(1, rgb_t(69, 62, 66));
+}
+
+void roland_d70_state::d70_map(address_map &map) {
+  map(0x0100, 0x0100).w(FUNC(roland_d70_state::bank_w));
+  map(0x0400, 0x07ff).rw(FUNC(roland_d70_state::ksga_io_r), FUNC(roland_d70_state::ksga_io_w));
+  map(0x0800, 0x0800).rw(FUNC(roland_d70_state::lcd_ctrl_r), FUNC(roland_d70_state::lcd_ctrl_w));
```

Comment:
For cases like this where an 8-bit peripheral is on one byte lane of a 16-bit data bus you can map it directly and add .umask16(0xff00) or .umask16(0x00ff) to choose which lane it's on.  Then you don't need the trampoline function.